client/types/types.go

```diff
@@ -91,6 +93,74 @@ func (o *Operation) Equal(o2 *Operation) error {
 	return nil
 }
 
+func (o *Operation) Filename() (filename string) {
+	filename = fmt.Sprintf("dkg_id_%s", o.DKGIdentifier[:5])
+
+	if strings.HasPrefix(string(o.Type), "state_signing_") {
+		var payload responses.SigningPartialSignsParticipantInvitationsResponse
+
+		if err := json.Unmarshal(o.Payload, &payload); err == nil {
+			filename = fmt.Sprintf("%s_signing_id_%s", filename, payload.BatchID)
+		}
+	}
+
+	return fmt.Sprintf(
+		"%s_step_%d_%s_%s.json",
+		filename,
+		getStepNumber(o.Type),
+		getShortOperationDescription(o.Type),
+		o.ID[:5],
+	)
+}
+
+func getShortOperationDescription(operationType OperationType) string {
+	switch fsm.State(operationType) {
+	case signature_proposal_fsm.StateAwaitParticipantsConfirmations:
+		return "confirm_participation"
+	case dkg_proposal_fsm.StateDkgCommitsAwaitConfirmations:
+		return "send_commits_for_the_DKG_round"
+	case dkg_proposal_fsm.StateDkgDealsAwaitConfirmations:
+		return "send_deals_for_the_DKG_round"
+	case dkg_proposal_fsm.StateDkgResponsesAwaitConfirmations:
+		return "send_responses_for_the_DKG_round"
+	case dkg_proposal_fsm.StateDkgMasterKeyAwaitConfirmations:
+		return "reconstruct_the_public_key_and_broadcast_it"
+	case signing_proposal_fsm.StateSigningAwaitPartialSigns:
+		return "partial_sign"
+	case signing_proposal_fsm.StateSigningPartialSignsCollected:
+		return "recover_full_signature"
+	case ReinitDKG:
+		return "reinit_DKG"
+	default:
+		return "unknown_operation"
+	}
+}
+
+func getStepNumber(operationType OperationType) int {
+	switch fsm.State(operationType) {
+	case dkg_proposal_fsm.StateDkgCommitsAwaitConfirmations:
+		return 1
+	case dkg_proposal_fsm.StateDkgDealsAwaitConfirmations:
+		return 2
+	case dkg_proposal_fsm.StateDkgResponsesAwaitConfirmations:
+		return 3
+	case dkg_proposal_fsm.StateDkgMasterKeyAwaitConfirmations:
+		return 4
+
+	case signature_proposal_fsm.StateAwaitParticipantsConfirmations:
+		return 1
+	case signing_proposal_fsm.StateSigningAwaitPartialSigns:
+		return 2
+	case signing_proposal_fsm.StateSigningPartialSignsCollected:
```

Comment:
this state leads straight to the idle state, it doesn't provide any available operations, thus you can remove it

Reply:
yes. but before internal switching, the fsm creates an operation https://github.com/lidofinance/dc4bc/blob/remove_qr/client/services/node/node_service.go#L746 with the state, that's why we got `dkg_id_7dfa1_signing_id_078dd012-eda5-4c15-bd05-959063470249_step_-1_recover_full_signature_70e70_response.json`
`-1` in today's corridor test
i would consider to return this step back)

Reply:
thanks! you're totally right, my mistake